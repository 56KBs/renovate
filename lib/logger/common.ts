export interface Logger {
  trace(msg: string): void;
  trace(meta: Record<string, any>, msg?: string): void;
  debug(msg: string): void;
  debug(meta: Record<string, any>, msg?: string): void;
  info(msg: string): void;
  info(meta: Record<string, any>, msg?: string): void;
  warn(msg: string): void;
  warn(meta: Record<string, any>, msg?: string): void;
  error(msg: string): void;
  error(meta: Record<string, any>, msg?: string): void;
  fatal(msg: string): void;
  fatal(meta: Record<string, any>, msg?: string): void;
}
