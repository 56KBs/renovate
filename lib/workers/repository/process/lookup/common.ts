import { UpdateType } from '../../../../config';

export interface LookupUpdate {
  blockedByPin?: boolean;
  branchName?: string;
  canBeUnpublished?: boolean;
  checksumUrl?: string;
  commitMessageAction?: string;
  displayFrom?: string;
  displayTo?: string;
  downloadUrl?: string;
  isBump?: boolean;
  isLockfileUpdate?: boolean;
  isPin?: boolean;
  isRange?: boolean;
  isRollback?: boolean;
  isSingleVersion?: boolean;
  fromVersion?: string;
  newDigest?: string;
  newDigestShort?: string;
  newMajor?: number;
  newMinor?: number;
  newValue: string;
  newVersion?: string;
  releaseTimestamp?: string;
  semanticCommitType?: string;
  toVersion?: string;
  updateType: UpdateType;
}
