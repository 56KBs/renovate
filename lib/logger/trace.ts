import Formatter from 'njstrace/lib/formatter';
import { Logger } from './common';

interface FormatterArgs {
  name: string;
  file: string;
  line: number;
  args: any[];
  stack: any[];
}

export class RenovateFormatter extends Formatter {
  logger: Logger;

  constructor(logger: Logger) {
    super();
    this.logger = logger;
  }

  onEntry(args: FormatterArgs): void {
    const params = args.args.map(arg => JSON.stringify(arg, null, 2)).join(',');
    this.logger.trace(`${args.name}(${params})`);
  }

  // eslint-disable-next-line class-methods-use-this
  onExit(args: FormatterArgs): void {}
}
