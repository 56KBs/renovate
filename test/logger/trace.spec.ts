import { RenovateFormatter } from '../../lib/logger/trace';
import { Logger } from '../../lib/logger/common';

let logged: string;
let logger: Logger = {} as any;

describe('trace', () => {
  beforeEach(() => {
    logged = null;
    logger = {
      trace(args: any) {
        logged = args;
      },
    } as any;
  });
  it('inits', () => {
    expect(new RenovateFormatter(null)).toBeDefined();
  });
  it('onEntry', () => {
    const formatter = new RenovateFormatter(logger);
    formatter.onEntry({ name: 'method', args: ['arg1', 'arg2'] });
    expect(logged).toMatchSnapshot();
  });
  it('onExit', () => {
    const formatter = new RenovateFormatter(logger);
    formatter.onExit({ name: 'method', args: ['arg1', 'arg2'] });
    expect(logged).toMatchSnapshot();
  });
});
