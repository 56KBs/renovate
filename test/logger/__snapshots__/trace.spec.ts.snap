// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trace onEntry 1`] = `"method(\\"arg1\\",\\"arg2\\")"`;

exports[`trace onEntry 2`] = `null`;
